lms/djangoapps/courseware/date_summary.py

```diff
@@ -405,9 +405,11 @@ def verified_upgrade_deadline_link(user, course=None, course_id=None):
 
     ecommerce_service = EcommerceService()
     if ecommerce_service.is_enabled(user):
+        course_mode = None
         if course is not None and isinstance(course, CourseOverview):
-            course_mode = course.modes.get(mode_slug=CourseMode.VERIFIED)
-        else:
+            course_mode = course.get_course_mode(CourseMode.VERIFIED)
```

Comment:
This line was causing another query before this change. The way I have it here removed the extra query since it uses the prefetched data.